src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/IndexStateManagementRestTestCase.kt

```diff
@@ -208,4 +217,55 @@ abstract class IndexStateManagementRestTestCase : ESRestTestCase() {
         val indexSettings = getIndexSettings(indexName) as Map<String, Map<String, Map<String, Any?>>>
         return indexSettings[indexName]!!["settings"]!!["index.blocks.write"] as String
     }
+
+    @Suppress("UNCHECKED_CAST")
+    protected fun getUuid(indexName: String): String {
+        val indexSettings = getIndexSettings(indexName) as Map<String, Map<String, Map<String, Any?>>>
+        return indexSettings[indexName]!!["settings"]!!["index.uuid"] as String
+    }
+
+    /**
+     * Compares responses returned by APIs such as those defined in [RetryFailedManagedIndexAction] and [RestAddPolicyAction]
+     *
+     * Example response with no failures:
+     * {
+     *   "failures": false,
+     *   "updated_indices": 3
+     * }
+     *
+     * Example response with failures:
+     * {
+     *   "failures": true,
+     *   "failed_indices": [
+     *     {
+     *       "index_name": "indexName",
+     *       "index_uuid": "s1PvTKzaThWoeA43eTPYxQ"
+     *       "reason": "Reason for failure"
+     *     }
+     *   ]
+     * }
+     */
+    @Suppress("UNCHECKED_CAST")
+    protected fun assertApiResponseIsEqual(expected: Map<String, Any>, actual: Map<String, Any>) {
```

Comment:
"assertApiResponseIsEqual" seems like we are just comparing any generic responses, but this is specific to ones that return in the structure outlined in the comment. Either rename or make this generic.

Reply:
Looked into how ```assertEquals``` behaves and it turns out that since this map response contains either primitive values or Strings and because ```Assert``` has a method ```internalArrayEquals``` which allows two arrays to be compared based on content, we can actually do ```assertEquals(expected, actual)``` on these two response maps.

However, this would require making sure the list of failed indices in the two response maps are in the same order. I began writing a utility method to do this for the "actual" message that we currently retrieve by doing ```response.asMap()``` but thought since this would mean the person writing the tests would need to write the "expectedMessage" response in order, that it's not very intuitive. So I will keep this assertion method and call it something more descriptive.